FreydCategoriesForCAP/gap/FreydCategoriesDerivedMethods.gi

```diff
@@ -368,11 +370,19 @@ AddFinalDerivationBundle( # WeakBiFiberProductMorphismToDirectSum,
 [
   WeakBiFiberProductMorphismToDirectSum,
   function( cat, alpha, beta )
-    local diagonal_difference;
+    local diagram, direct_sum_diagram, direct_sum, list_of_morphisms, joint_pairwise_difference;
     
-    diagonal_difference := DirectSumDiagonalDifference( cat, [ alpha, beta ] );
+    diagram := [ alpha, beta ];
     
-    return WeakKernelEmbedding( cat, diagonal_difference );
+    direct_sum_diagram := List( diagram, Source );
+    
+    direct_sum := DirectSum( cat, direct_sum_diagram );
+    
+    list_of_morphisms := List( [ 1 .. 2 ], i -> PreCompose( cat, ProjectionInFactorOfDirectSumWithGivenDirectSum( cat, direct_sum_diagram, i, direct_sum ), diagram[ i ] ) );
+    
+    joint_pairwise_difference := JointPairwiseDifferencesOfMorphismsIntoDirectProduct( cat, direct_sum, list_of_morphisms );
```

Comment:
I suggest spelling out everything explicitly here, i.e. getting rid of lists and `JointPairwiseDifferencesOfMorphismsIntoDirectProduct` and simply subtracting the two projections composed with `alpha` and `beta`.